letsencrypt-auto-source/letsencrypt-auto.template

```diff
@@ -358,7 +358,7 @@ if [ "$1" = "--le-auto-phase2" ]; then
   # If the venv doesn't exist, we just bootstrapped
   if [ -d "$VENV_PATH" -a "$BOOTSTRAP_VERSION" != "$(PrevBootstrapVersion)" ]; then
     # if non-interactive mode or stdin and stdout are connected to a terminal
-    if [ "$NONINTERACTIVE" = 1 -o -t 0 -a -t 1]; then
+    if [ \( "$NONINTERACTIVE" = 1 \) -o \( \( -t 0 \) -a \( -t 1 \) \) ]; then
```

Comment:
Interesting. Are the parens necessary? I was going to ask you to put them in to help poor people like me who don't know the relative precedence of -a and -o by heart, but then I couldn't find any docs on parens, and "seems to work" is a lousy guide for portable software. :-)